modules/virtualisation/podman/podman.nix

```diff
@@ -0,0 +1,183 @@
+      type = types.bool;
+      default = false;
+      description = ''
+        Create an alias mapping <command>docker</command> to <command>podman</command>.
+      '';
+    };
+
+    enableNvidia = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Enable use of NVidia GPUs from within podman containers.
+      '';
+    };
+
+    extraPackages = mkOption {
+      type = with types; listOf package;
+      default = [ ];
+      example = lib.literalExample ''
+        [
+          pkgs.gvisor
+        ]
+      '';
+      description = ''
+        Extra packages to be installed in the Podman wrapper.
+      '';
+    };
+
+    package = lib.mkOption {
+      type = types.package;
+      default = podmanPackage;
+      internal = true;
+      description = ''
+        The final Podman package (including extra packages).
+      '';
+    };
+
+    defaultNetwork.extraPlugins = lib.mkOption {
+      type = types.listOf json.type;
+      default = [ ];
+      description = ''
+        Extra CNI plugin configurations to add to podman's default network.
+      '';
+    };
+
+  };
+
+  config = lib.mkIf cfg.enable (lib.mkMerge [
+    {
+      home.packages = [ cfg.package ]
+        ++ lib.optional cfg.dockerCompat dockerCompat;
+
+      xdg.configFile."cni/net.d/87-podman-bridge.conflist".source =
+        net-conflist;
+
+      virtualisation.containers = {
+        enable = true; # Enable common /etc/containers configuration
+        containersConf.settings = lib.optionalAttrs cfg.enableNvidia {
+          engine = {
+            conmon_env_vars =
+              [ "PATH=${lib.makeBinPath [ pkgs.nvidia-podman ]}" ];
+            runtimes.nvidia =
+              [ "${pkgs.nvidia-podman}/bin/nvidia-container-runtime" ];
+          };
+        };
+      };
+
+      systemd.user = {
+
+        services.podman = {
+          Unit = {
+            Description = "Podman API Service";
+            Requires = "podman.socket";
+            After = "podman.socket";
+            Documentation = "man:podman-system-service(1)";
+            StartLimitIntervalSec = 0;
+          };
+          Service = {
+            Type = "exec";
+            KillMode = "process";
+            Environment = ''LOGGING=" --log-level=info"'';
+            ExecStart =
+              [ "" "${cfg.package}/bin/podman $LOGGING system service" ];
+          };
+
+          Install = { WantedBy = [ "multi-user.target" ]; };
+        };
+
+        sockets.podman = {
+          Unit = {
+            Description = "Podman API Socket";
+            Documentation = "man:podman-system-service(1)";
+          };
+          Socket = {
+            ListenStream = "%t/podman/podman.sock";
+            SocketMode = 660;
+          };
+          Install.WantedBy = [ "sockets.target" ];
+        };
+
+      };
+    }
+    (lib.mkIf cfg.dockerSocket.enable {
+      home.sessionVariables."DOCKER_HOST" =
+        "unix:///run/user/$UID/podman/podman.sock";
```

Comment:
I'm really not sure about how to do this properly?

Reply:
`$XDG_RUNTIME_DIR` is the variable used for this.

Variables in `home.sessionVariables` are not expanded (AFAIK) so this would need to be added to `home.sessionVariablesExtra` (which is a string).

Reply:
The variable seems to get expanded for me, I'm not sure if that's a bug